redwoods-insurance-salesforce/force-app/main/default/lwc/incidentMap/__tests__/incidentMap.test.js

```diff
@@ -0,0 +1,75 @@
+import { createElement } from 'lwc';
+import IncidentMap from 'c/incidentMap';
+import { getRecord } from 'lightning/uiRecordApi';
+import { registerLdsTestWireAdapter } from '@salesforce/sfdx-lwc-jest';
+
+const getRecordAdapter = registerLdsTestWireAdapter(getRecord);
+const mockMapMarkers = require('./data/mockMapMarkers.json');
+
+describe('c-incident-map', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    it('invokes the wire adapter with default properties', () => {
+        // Create initial element
+        const element = createElement('c-incident-map', {
+            is: IncidentMap
+        });
+        element.recordId = 'mockRecordId';
+        document.body.appendChild(element);
+
+        return Promise.resolve().then(() => {
+            expect(getRecordAdapter.getLastConfig()).toEqual({
+                recordId: 'mockRecordId',
+                fields: [
+                    'Case.Incident_Location__Latitude__s',
+                    'Case.Incident_Location__Longitude__s'
+                ]
+            });
+        });
+    });
+
+    it('does not render the map component when no map markers are set', () => {
+        // Create initial element
+        const element = createElement('c-incident-map', {
+            is: IncidentMap
+        });
+        document.body.appendChild(element);
+
+        const mapEl = element.shadowRoot.querySelector('lightning-card');
+        expect(mapEl).toBe(null);
+    });
+
+    it('renders the map compoent when the wire adapter generates a map marker object', () => {
+        // Create initial element
+        const element = createElement('c-incident-map', {
+            is: IncidentMap
+        });
+        document.body.appendChild(element);
+        getRecordAdapter.emit(mockMapMarkers);
+
+        return Promise.resolve().then(() => {
+            //const mapEl = element.shadowRoot.querySelector('lightning-card');
+            const mapEl = element.shadowRoot
+                .querySelector('lightning-card')
```

Comment:
You don't have to query for the `lightning-card` here, as `lightning-map` is part of the components DOM.

Reply:
You're wrong. I worked with @Tbliss for an hour or so yesterday to get this test working. Without chaining the calls together Its not able to find a lightning-map component. This was his suggestion, and the the only one that worked that allowed me to assert off the presence of an actual map element, instead of it's parent card.

Reply:
Nope, not wrong. Every single recipe in LWC Recipes has as the most outer container `lightning-card`. Please remove.

Reply:
@muenzpraeger You're comment to check the map-markers attribute actually led to the discovery that this test is a false-positive-pass. The const MapEl, as selected returns an empty `{}` and not anything with attributes to test. Interestingly enough, simply querying for the the 'lightning-card' parent, also returns an empty `{}` which means I'm at a loss as to how one gets proper access to a component nested inside another (like a card). Please advise.

Reply:
`const mapEl = element.shadowRoot.querySelector('lightning-map')`

Reply:
@muenzpraeger then you're going to have to do more than just claim 'every other recipe works' because I've been cribbing from lwc-recipes, and in this case it's not working. Pointing people at other code is not a pedagogical tool when they're obviously missing some nuanced bit from the code they've already reviewed.

Reply:
@muenzpraeger do you not think I've tried that? (I have). that returns an empty object `{}` again, code without explanation does not help

Reply:
See here.

![image](https://user-images.githubusercontent.com/2205451/87559923-2fb81a80-c6bb-11ea-8fd5-e0770d8e5890.png)

Reply:
@muenzpraeger Current code:

``` js
return Promise.resolve().then(() => {
            const mapEl = element.shadowRoot
                //.querySelector('lightning-card')
                .querySelector('lightning-map');
            expect(mapEl).not.toBe(null);
            console.log(JSON.stringify(mapEl));
            //expect(mapEl).
        });
```

Output: 
```
 PASS  force-app/main/default/lwc/incidentMap/__tests__/incidentMap.test.js
  c-incident-map
    ✓ invokes the wire adapter with default properties (10ms)
    ✓ does not render the map component when no map markers are set (6ms)
    ✓ renders the map compoent when the wire adapter generates a map marker object (30ms)
    ✓ does not render the map when an error occurs (1ms)

  console.log
    {}

      at force-app/main/default/lwc/incidentMap/__tests__/incidentMap.test.js:60:13

```

as the console.log notes, the lightning-map element is an empty object.

Reply:
@muenzpraeger I just ran your exact code, and can duplicate your results, but I need to understand why. Jest is doing some kind of unintuitive magic here, and I wouldn't have even _guessed_ that would have worked.

Reply:
@codefriar Jest has nothing to do with that, it's likely because of LWC. `mapEl` is not a simple `HTMLElement`, it's a `HTMLBridgeElement` and it may be protected to simulate Shadow DOM.